src/nunit.analyzers.tests/TestCaseUsage/TestCaseUsageAnalyzerTests.cs

```diff
@@ -755,5 +755,60 @@ public sealed class AnalyzeWhenTestMethodHasTypeParameterArgumentType
     }");
             RoslynAssert.Valid(this.analyzer, testCode);
         }
+
+#if NUNIT4
```

Comment:
Do we need this when we have declared the CancelAfterAttribute for NUNIT3 ?

Reply:
Yes, the test project has this class only because of the `NUnitFrameworkConstantsTests` class.
These test here are compiled against the real NUnit assembly which for NUnit 3 doesn't have the `CancelAfterAttribute`.